include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -492,7 +492,13 @@ namespace Differentiation
      * to simplify the final substitution map by resolving all explicit
      * interdependencies between entries in the substitution map.
      */
-    template <typename ExpressionType = SD::Expression, typename ValueType>
+    template <typename ExpressionType = SD::Expression,
+              typename ValueType,
+              typename = typename std::enable_if<
+                dealii::internal::is_explicitly_convertible<
+                  ExpressionType,
+                  const SymEngine::RCP<const SymEngine::Basic> &>::value &&
+                std::is_constructible<ExpressionType, ValueType>::value>::type>
```

Comment:
Wow. Might want to put a human-readable version of this condition into the documentation :-)